src/plugins/plugin-http.ts

```diff
@@ -44,118 +44,139 @@ function extractUrl(options) {
     (options.path || options.pathName || '/');
 }
 
-function unpatchHttp(http) {
-  shimmer.unwrap(http, 'request');
-  if (semver.satisfies(process.version, '>=8.0.0')) {
-    shimmer.unwrap(http, 'get');
-  }
+function isTraceAgentRequest (options, api) {
+  return options && options.headers &&
+    !!options.headers[api.constants.TRACE_AGENT_REQUEST_HEADER];
+}
+
+function makeRequestTrace(request, api) {
```

Comment:
This code reorganization makes it harder to review. Is anything changed in this file?

Reply:
`makeRequestTrace` and `isTraceAgentRequest` are unchanged except they now each take in an `api` object. They were taken out of `patchHttp` so they can be shared with `patchHttps`.